templates/client-daemonset.yaml

```diff
@@ -205,11 +205,18 @@ spec:
                 {{- if (and .Values.global.gossipEncryption.secretName .Values.global.gossipEncryption.secretKey) }}
                 -encrypt="${GOSSIP_KEY}" \
                 {{- end }}
+                {{- if or .Values.client.join (and .Values.externalServers.enabled .Values.externalServers.hosts) }}
                 {{- if .Values.client.join }}
                 {{- range $value := .Values.client.join }}
                 -retry-join={{ quote $value }} \
                 {{- end }}
                 {{- else }}
+                {{- $serfPort := (.Values.externalServers.serfLANPort | int) -}}
+                {{- range $host := .Values.externalServers.hosts }}
+                -retry-join={{ printf "%s:%d" $host $serfPort | quote}} \
```

Comment:
Does this still work if someone set a provider? Or a dns name? I think the [docs](https://www.consul.io/docs/agent/options.html#retry-join) also mention being able to set a go-sockaddr template.